### ISSUE TITLE
Nhóm mình đã hoàn thành và có thể góp ý chỉnh sửa

### ISSUE THREAD

**Issue by SyThuan1803** [SyThuan1803/Data_Science_Proramming_FinalProject#4]

**Comment by robocon20x** [comment]
- dễ bị outliner khi random 1000 dòng nếu đã dùng độ tương tự thì số lượng lấy có ít quá không khi có tận 7000 dòng cần điền mà chỉ thêm 1000 dòng có giá trị? Mình nghĩ có thể bỏ những dòng không có kết quả đánh giá hài lòng, những dòng khác sắp xếp theo mức độ hài lòng và chọn những dòng điền full có tỉ lệ tương tự như tỉ lệ các cột chưa điền full dòng theo mức độ hài lòng.
- Câu 3: có nhất thiết phải xây dựng cây phân loại không trong khi chỉ cần tính entropy đầu tiên để biết được độ quan trọng của các yếu tố ảnh hưởng đến kết quả cuối? Bởi vì cây phân loại được xây theo độ quan trọng của các thuộc tính sau theo thuộc tính trước chứ không phải cứ 5 thuộc tính ở 5 độ sâu đầu của cây phân loại là 5 thuộc tính quan trọng nhất

**Comment by npqd** [comment]
- Theo mình nghĩ sửa lại mấy cái cell markdown bị lỗi cái phần xuống dòng với cell danh sách thành viên lỗi khi up lên github ấy. Với lại nên up file đã xuất output để đỡ phải clone về để run lại ấy.
- Câu 3: theo mình xuất cây quyết định cũng không cần thiết lắm, vì này khá nhiều cột dữ liệu và cây bạn in ra thì khoãng 370 dòng theo mình đếm thì này làm người đọc khá rối và dư thừa. Hoặc nếu muốn có vài thư viện để vẽ cây ra "dễ" nhìn hơn chút xíu có thể tham khảo chứ cây này mình hơi rối @@!

**Comment by SyThuan1803** [comment]
Cảm ơn góp ý của 2 bạn nha. Nhóm mình sẽ xem xét và chỉnh sửa. 
Mà mình giải thích xíu ở câu 1 là do nhiều dòng dữ liệu quá, thời gian chạy lâu nên mình chọn lấy mẫu 1000 dòng, nếu mà có thời gian thì mình sẽ chạy hết luôn.
Còn câu 3 thì việc xếp hạng độ quan trọng của các thuộc tính ở đây mình dùng độ đo RFE (recursive feature elimination) cho thuật toán phân lớp là cây quyết định. RFE sẽ chọn ra n thuộc tính quan trọng nhất (max information gain). Link tham khảo: https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.RFE.html#sklearn.feature_selection.RFE. 
Kết quả do RFE chọn ra cũng giống như 5 thuộc tính đầu được chia trong cây quyết định. Và theo mình biết thì thuật toán ID3 sẽ chọn ra thuộc tính có max information gain để chia, rồi từ đó lại chọn ra thuộc tính khác có max information gain, cho nên việc tính entropy từ đầu như vậy là không đủ. 
Nhóm mình cũng tạo mô hình cây quyết định vì có thể sau này sẽ dùng để phân lớp, và việc áp dụng cây quyết định cũng không ảnh hưởng quá nhiều thời gian chạy notebook.

**Comment by robocon20x** [comment]
> 

chào bạn, theo mình có tìm hiểu về RFE thì nó sẽ lần lượt thử các thuộc tính có trong  số hoán vị 1...n phần tử, và chọn ra số phần tử nào cho kết quả accuracy tốt nhất và chọn các phần tử của tập hoán vị đó. Nếu kết quả là như thế mình nghĩ bạn nên đặt lại câu hỏi là nhóm thuộc tính nào quan trọng nhất ảnh hưởng đến mức độ hài lòng. Còn nếu chỉ trả lời cho câu hỏi những thuộc tính có ảnh hưởng lớn nhất tới độ hài lòng, mình nghĩ sẽ là tính entropy của tất cả các thuộc tính và chọn ra 5 cái có kết quả cao nhất, vì xét cho cùng cây phân loại thì thuộc tính đi sau luôn luôn phải tính dựa theo các thuộc tính là cha của nó. Nhưng dù sao thì bài của nhóm bạn làm quá hoàn chỉnh, áp dụng hết kiến thức đã học của bộ môn, không làm quá sơ sài như nhóm mình do tìm được tập dữ liệu có đầy đủ các thông tin cần thiết.
